Sources/SecureXPC/Server/XPCServiceServer.swift

```diff
@@ -43,10 +43,12 @@ internal class XPCServiceServer: XPCServer {
     
 	public override func startAndBlock() -> Never {
 		xpc_main { connection in
+            xpc_connection_set_target_queue(connection, XPCServiceServer.service.targetQueue)
 			// Listen for events (messages or errors) coming from this connection
 			xpc_connection_set_event_handler(connection, { event in
 				XPCServiceServer.service.handleEvent(connection: connection, event: event)
 			})
+            XPCServiceServer.service.connections.append(WeakConnection(connection))
```

Comment:
Should the connections be removed from this pool upon invalidation/interruption/inwhatever?

Reply:
I believe the connections themselves won't be held as it's only a weak reference? So the array over time would grow to have more weak wrappers that have a `nil` connection reference, which certainly isn't ideal. Hmm, will give this a think of how to do this in a simple manner.

Reply:
Okay I've figured out how to do this by using `xpc_connection_set_finalizer_f` and confirmed it works as I expected. However, to do this in a simple way I actually expanded onto `WeakConnection` so that it's rather specialized now and holds onto a reference of its containing server.

Reply:
Oh that's a great idea.